cmd/cost/get.go

```diff
@@ -0,0 +1,230 @@
+			ParentId:  OU.Id,
+			NextToken: nextToken,
+		})
+		if err != nil {
+			log.Fatalln("Unable to retrieve child OUs under OU:", err)
+		}
+
+		//Add OUs to slice
+		for childOU := 0; childOU < len(OUs.OrganizationalUnits); childOU++ {
+			OUSlice = append(OUSlice, OUs.OrganizationalUnits[childOU])
+		}
+
+		if OUs.NextToken == nil {
+			break
+		}
+		nextToken = OUs.NextToken //If NextToken != nil, keep looping
+	}
+
+	return OUSlice
+}
+
+//Get the account IDs of all (not only immediate) accounts under OU
+func getOUsRecursive(OU *organizations.OrganizationalUnit, awsClient awsprovider.Client) []*organizations.OrganizationalUnit {
+	var OUs []*organizations.OrganizationalUnit
+
+	//Populate OUs by getting immediate OUs (direct nodes)
+	currentOUs := getOUs(OU, awsClient)
+
+	//Loop through all child OUs. Append the child OU, then append the OUs of the child OU
+	for _, currentOU := range currentOUs {
+		OUs = append(OUs, currentOU)
+		OUs = append(OUs, getOUsRecursive(currentOU, awsClient)...)
+	}
+
+	return OUs
+}
+
+//Get cost of given account
+func getAccountCost(accountID *string, awsClient awsprovider.Client, timePtr *string, cost *float64) {
+	//Starting from the 1st of the current month last year i.e. if today is 2020-06-29, then start date is 2019-06-01
+	start := strconv.Itoa(time.Now().Year()-1) + time.Now().Format("-01-") + "01"
+	end := time.Now().Format("2006-01-02")
+	granularity := "MONTHLY"
+	metrics := []string{
+		"NetUnblendedCost",
+	}
+
+	switch *timePtr {
+	case "MTD":
+		start = time.Now().Format("2006-01") + "-01"
+		end = time.Now().Format("2006-01-02")
+	case "YTD":
+		start = time.Now().Format("2006") + "-01-01"
+		end = time.Now().Format("2006-01-02")
+	case "TestError":
+		start = "2020-05-23"
+		end = "2019-06-12"
+	}
+
+	//Get cost information for chosen account
+	costs, err := awsClient.GetCostAndUsage(&costexplorer.GetCostAndUsageInput{
+		Filter: &costexplorer.Expression{
+			Dimensions: &costexplorer.DimensionValues{
+				Key: aws.String("LINKED_ACCOUNT"),
+				Values: []*string{
+					accountID,
+				},
+			},
+		},
+		TimePeriod: &costexplorer.DateInterval{
+			Start: aws.String(start),
+			End:   aws.String(end),
+		},
+		Granularity: aws.String(granularity),
+		Metrics:     aws.StringSlice(metrics),
+	})
+	if err != nil {
+		log.Fatalln("Error getting costs report:", err)
+	}
+
+	//Loop through month-by-month cost and increment to get total cost
+	for month := 0; month < len(costs.ResultsByTime); month++ {
+		monthCost, err := strconv.ParseFloat(*costs.ResultsByTime[month].Total["NetUnblendedCost"].Amount, 64)
+		if err != nil {
+			log.Fatalln("Unable to get cost:", err)
+		}
+		*cost += monthCost
+	}
+}
+
+//Get cost of given OU by aggregating costs of only immediate accounts under given OU
+func getOUCost(OU *organizations.OrganizationalUnit, awsClient awsprovider.Client, timePtr *string, cost *float64) {
```

Comment:
For all of the `getXCost`, why not return a float64 instead of having to deal with a pointer?

Reply:
My initial thought was that it's neater to use pointers, as it saves a line for returning 😅. Also probably overthinking this, but passing and copying a pointer as opposed to a float64 might be more efficient for older architectures.

Do you think I should return instead to make it more intuitive, or?

Reply:
For me, it adds a bit of confusion. If I call `getCost` I expect a return of some number. If I was going to use a pointer (personally) I would have it as the first argument because that's the value I'm really interested in.

Reply:
Thanks for the advice, I think that would be really helpful for everyone. I've changed the pointer to the first argument.

Reply:
:+1: